x/evm/keeper/scheduler_job.go

```diff
@@ -63,17 +66,55 @@ func (k Keeper) ExecuteJob(ctx sdk.Context, definition, payload []byte, senderPu
 	if err != nil {
 		return err
 	}
+
+	var hexBytes []byte
+	switch {
+	case senderAddress != nil:
+		hexBytes, err = hex.DecodeString(addressToHex(senderAddress))
```

Comment:
What is this pair of methods doing because it looks like it just does `unhex(hex(blob))`. Why does that not mean `hexBytes := blob`?